components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,59 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+// Helper function which returns true if the runtime state is trackable
+func isTrackableState(state kebruntime.State) bool {
+	switch state {
+	case kebruntime.StateSucceeded, kebruntime.StateError, kebruntime.StateUpgrading, kebruntime.StateUpdating:
 		return true
 	}
 	return false
 }
 
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
+	// Check if the cluster is in a trackable state
+	if isTrackableState(runtime.Status.State) ||
+		(runtime.Status.Provisioning != nil &&
+			runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
+			runtime.Status.Deprovisioning == nil) {
```

Comment:
This is hard to read, could you please encapsulate this logic in a function?
From the logic I read, the function signature could be `func isProvisioned(runtime) bool`, what do you think?

Reply:
Done